fvm/environment/derived_data_invalidator.go

```diff
@@ -22,10 +22,47 @@ type ContractUpdate struct {
 type DerivedDataInvalidator struct {
 	ContractUpdateKeys []ContractUpdateKey
 	FrozenAccounts     []common.Address
+
+	MeterParamOverridesUpdated bool
 }
 
 var _ derived.TransactionInvalidator = DerivedDataInvalidator{}
 
+func NewDerivedDataInvalidator(
+	contractKeys []ContractUpdateKey,
+	env *facadeEnvironment,
+) DerivedDataInvalidator {
+	return DerivedDataInvalidator{
+		ContractUpdateKeys:         contractKeys,
+		FrozenAccounts:             env.FrozenAccounts(),
+		MeterParamOverridesUpdated: meterParamOverridesUpdated(env),
+	}
+}
+
+func meterParamOverridesUpdated(env *facadeEnvironment) bool {
+	updatedRegisterIds, _ := env.txnState.RegisterUpdates()
+
+	serviceAccount := string(env.chain.ServiceAddress().Bytes())
+	storageDomain := common.PathDomainStorage.Identifier()
+	for _, registerId := range updatedRegisterIds {
+		if registerId.Owner != serviceAccount {
+			continue
+		}
+
+		// NOTE: This condition is empirically generated by running the
+		// MeterParamOverridesComputer to capture touched registers.
+		// The parameter settings are store in the service account's slabs.
```

Comment:
This only explains why we check `state.IsSlabIndex(registerId.Key)` to me. 

Could we also explain why we also check `registerId.Key == storageDomain`?

Reply:
(imo, slabs/storageDomain key, are a leaky abstraction that should not be exposed to cadence in the first place.  we should have provided register ids ordering guarantees instead and let the ledger handle the storage layout internally, but i digress ...)

Most, but not all, values/fields in an account are stored in an ordered map known only to cadence.  cadence encodes the map into bytes and split the bytes into slab chunks, then stores the slab chunks into the ledger.  When we check for stab id, we're basically checking for any changes to the ordered map.

I don't really know what's stored in `storageDomain` but that's one of the input for computing the meter param overrides and hence any changes to it will also invalidate the derived value.

I've tweak the comments a bit